stacks/sscheck_backend_stack.py

```diff
@@ -0,0 +1,151 @@
+    aws_route53 as route53,
+    aws_route53_targets as route53t,
+    aws_certificatemanager as acm
+)
+
+class SampleSheetCheckBackendStack(cdk.Stack):
+
+    def __init__(self, scope: cdk.Construct, construct_id: str, **kwargs) -> None:
+        super().__init__(scope, construct_id, **kwargs)
+
+        # --- Cognito parameters are from data portal terraform stack
+        cog_user_pool_id = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "CogUserPoolID",
+            string_parameter_name="/data_portal/client/cog_user_pool_id",
+        ).string_value
+
+        # Load SSM parameter (created Via Console)
+        data_portal_metadata_api = ssm.StringParameter.from_string_parameter_attributes(self, "urlValue",
+            parameter_name="/sscheck/metadata-api"
+        ).string_value
+
+        # Query domain_name config from SSM Parameter Store (Created via Conosle)
+        domain_name = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "DomainName",
+            string_parameter_name="/sscheck/domain",
+        ).string_value
+
+        # --- Query deployment env specific config from SSM Parameter Store
+        hosted_zone_id = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "HostedZoneID",
+            string_parameter_name="hosted_zone_id"
+        ).string_value
+
+        hosted_zone_name = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "HostedZoneName",
+            string_parameter_name="hosted_zone_name"
+        ).string_value
+        
+        cert_use1_arn = ssm.StringParameter.from_string_parameter_name(
+            self,
+            "SSLCertUSE1ARN",
+            string_parameter_name="cert_use1_arn",
+        )
+
+        cert_use1 = acm.Certificate.from_certificate_arn(
+            self,
+            "SSLCertUSE1",
+            certificate_arn=cert_use1_arn.string_value,
+        )
+        # Create a Lambda Layer
+        sample_check_layer = lambda_.LayerVersion(
+            self,
+            "SamplecheckLambdaLayer",
+            code=lambda_.Code.from_asset("lambdas/layers/umccr_utils/python38-umccr_utils.zip"),
+            compatible_runtimes=[lambda_.Runtime.PYTHON_3_8],
+            description="A samplecheck library layer for python 3.8"
+        )
+
+        # Create a lambda function along with the layered crated above
+        sample_sheet_check_lambda = lambda_.Function(
+            self,
+            "SampleSheetValidationLambda",
+            runtime=lambda_.Runtime.PYTHON_3_8,
+            timeout=cdk.Duration.seconds(60),
+            code=lambda_.Code.from_asset("lambdas/functions"),
+            handler="main.lambda_handler",
+            layers=[sample_check_layer],
+            environment={"data_portal_metadata_api": data_portal_metadata_api}
+        )
+
+        # Cors Configuration
+        cors_config = apigateway.CorsOptions(
+            allow_origins = [ '*' ],
```

Comment:
This can be specific now. Pls change it to 

```
allow_origins = [ 'https://sscheck.dev.umccr.org', 'https://sscheck.prod.umccr.org', 'https://sscheck.umccr.org' ]
```

Reply:
Thanks for pointing that out. 
Reconfigured at ` commit 57247d4`

Will also add it on the backend as this uses the lambda integration